src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/MathFunction.cs

```diff
@@ -67,23 +68,172 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
 
             var arg = args[0];
             var argType = argTypes[0];
-            fValid &= CheckNumericColumnType(argType, arg, errors, ref nodeToCoercedTypeMap);
+            fValid &= CheckNumericColumnType(argType, arg, errors, ref nodeToCoercedTypeMap, context, out returnType);
 
-            if (nodeToCoercedTypeMap?.Any() ?? false)
-            {
-                // Now set the coerced type to a table with numeric column type with the same name as in the argument.
-                returnType = nodeToCoercedTypeMap[arg];
-            }
-            else
+            if (!fValid)
             {
-                returnType = argType;
+                nodeToCoercedTypeMap = null;
             }
 
-            returnType = context.Features.ConsistentOneColumnTableResult ? DType.CreateTable(new TypedName(DType.Number, GetOneColumnTableResultName(context.Features))) : returnType;
+            return fValid;
+        }
+    }
+
+    // Abstract base class for all 2-arg math functions that return numeric values.
+    internal abstract class MathTwoArgFunction : BuiltinFunction
+    {
+        public override ArgPreprocessor GetArgPreprocessor(int index)
+        {
+            return base.GetGenericArgPreprocessor(index);
+        }
 
-            if (!fValid)
+        public override bool IsSelfContained => true;
+
+        public MathTwoArgFunction(string name, TexlStrings.StringGetter description, int minArity)
```

Comment:
Same here, explicit syntax where this is being called, minArity: 1 etc

Reply:
Done.